src/snowflake/snowpark/_internal/data_source/dbms_dialects/base_dialect.py

```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from snowflake.snowpark.types import StructType
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class BaseDialect(ABC):
+    @abstractmethod
+    def generate_select_query(self, table_or_query: str, schema: StructType) -> str:
```

Comment:
can this be a static method since there is no member var of this BaseDialect and its child classes?
if so we do not even to instantiate the BaseDialect object, we can just use the static method on a class